test/common/upstream/host_stats_test.cc

```diff
@@ -0,0 +1,41 @@
+#include "envoy/upstream/host_description.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Upstream {
+namespace {
+
+// Verify that counters are sorted by name.
+TEST(HostStatsTest, CountersSortedByName) {
+  HostStats host_stats;
+  auto counters = host_stats.counters();
```

Comment:
the style guide discourages 'auto' in this case (the google public one); in this case I usually make a `using` nickname for the complex type being returned so you can give an explicit type here.

Just a suggestion.

Reply:
Done.